### DISCUSSION TITLE
Running chat-sdk through our own backend

### DISCUSSION THREAD

**Discussion by hiashutoshsingh** [GetStream/stream-chat-flutter#734] [Q&A]
My use case is that all the functionalities like sending and receive messages, channel fetching, channel creation, listening events etc. should be communicated from my backend and the backend will communicate to the get-stream server.

How can I achieve this?

I tried this 👇  but can't achieve it.

In `stream_chat_flutter_core` package we have `StreamChatClient` in which we pass streamApiKey and in this, we have an optional `chatApi` parameter which I can't seems to override as `StreamChatApi` class is not exposed, even if I import the `stream_chat` package separately.

**Comment by imtoori** [comment]
hey @hiashutoshsingh that's a weird use-case, but I think you can do something similar to this 
> In stream_chat_flutter_core package we have StreamChatClient in which we pass streamApiKey and in this, we have an optional chatApi parameter which I can't seems to override as StreamChatApi class is not exposed, even if I import the stream_chat package separately.
At the moment those classes are not exported, but you can still import them using a specific import like this one
`import 'package:stream_chat/src/core/api/stream_chat_api.dart';`
